### DISCUSSION TITLE
Integration with vimdiff

### DISCUSSION THREAD

**Discussion by antonmry** [CopilotC-Nvim/CopilotChat.nvim#700] [Ideas]
The new diff functionality is awesome. It would be very useful to open a new buffer with the proposed file if Copilot provides it. The only required thing would be to create a buffer with the file in a vertical split with the file proposed by Copilot.  With that, the user can do a :windo diffthis in the same way we do for git conflicts or other stuff.

I usually do that with the copilot buffer, it's a bit messy but it allows me to easily apply changes with `do` or `dp`:

<img width="1891" alt="image" src="https://github.com/user-attachments/assets/8340684c-a800-4584-a20a-3b8c43506e03" />

**Comment by deathbeam** [comment]
I was actually messing with implementing something like this, i think its good idea. also maybe this will be better as an issue than discussion? I can create one probs

**Reply by antonmry** [reply]
I wasn't sure if it would make sense but if you find it interesting, I've created an issue here in case it helps: https://github.com/CopilotC-Nvim/CopilotChat.nvim/issues/705 . Thanks for taking the feedback in consideration, this is a great plugin!
